### PR TITLE
Specify predictor version

### PR DESCRIPTION
**Pull request by skrakau**
See #36.
Used software version from `get_software_versions` to specify Fred2 predictor version and check if tool version is supported by Fred2 (see #11). 

#
Many thanks for contributing to nf-core/epitopeprediction!

Please fill in the appropriate checklist below (delete whatever is not relevant). These are the most common things requested on pull requests (PRs).

## PR checklist
 - [x] This comment contains a description of changes (with reason)
 - [ ] If you've fixed a bug or added code that should be tested, add tests!
 - [ ] If necessary, also make a PR on the [nf-core/epitopeprediction branch on the nf-core/test-datasets repo]( https://github.com/nf-core/test-datasets/pull/new/nf-core/epitopeprediction)
 - [ ] Ensure the test suite passes (`nextflow run . -profile test,docker`).
 - [ ] Make sure your code lints (`nf-core lint .`).
 - [ ] Documentation in `docs` is updated
 - [ ] `CHANGELOG.md` is updated
 - [ ] `README.md` is updated

**Learn more about contributing:** https://github.com/nf-core/epitopeprediction/tree/master/.github/CONTRIBUTING.md

### CODE REVIEW

**Comment by skrakau** [comment]
This has to wait for https://github.com/FRED-2/Fred2/pull/240

File: bin/epaa.py [hunk 1]
```diff
@@ -766,13 +763,12 @@ def make_predictions_from_variants(variants_all, methods, alleles, minlength, ma
...
-            for m in methods:
+            for method, version in methods.items():
                 try:
-                    predictor = EpitopePredictorFactory(m)
-                    tools_map[m] = '{name}-{version}'.format(name=m, version=predictor.version)
+                    predictor = EpitopePredictorFactory(method, version=version)
                     results.extend([predictor.predict(filtered_peptides, alleles=alleles)])
                 except:
-                    logging.warning("Prediction for length {length} and allele {allele} not possible with {method}.".format(length=peplen, allele=','.join([str(a) for a in alleles]), method=m))
+                    logging.warning("Prediction for length {length} and allele {allele} not possible with {method} v{version}.".format(length=peplen, allele=','.join([str(a) for a in alleles]), method=method, version=version))
```
Comment (christopher-mohr) [inline bin/epaa.py:771]: `with {method} version {version}`... ?

File: bin/epaa.py [hunk 2]
```diff
@@ -847,13 +840,12 @@ def make_predictions_from_peptides(peptides, methods, alleles, protein_db, ident
...
-        for m in methods:
+        for method, version in methods.items():
             try:
-                predictor = EpitopePredictorFactory(m)
-                tools_map[m] = '{name}-{version}'.format(name=m, version=predictor.version)
+                predictor = EpitopePredictorFactory(method, version=version)
                 results.extend([predictor.predict(all_peptides_filtered, alleles=alleles)])
             except:
-                logging.warning("Prediction for length {length} and allele {allele} not possible with {method}. No model available.".format(length=peplen, allele=','.join([str(a) for a in alleles]), method=m))
+                logging.warning("Prediction for length {length} and allele {allele} not possible with {method} v{version}. No model available.".format(length=peplen, allele=','.join([str(a) for a in alleles]), method=method, version=version))
```
Comment (christopher-mohr) [inline bin/epaa.py:848]: see above

File: bin/epaa.py [hunk 3]
```diff
@@ -909,6 +901,7 @@ def __main__():
     parser.add_argument('-l', "--max_length", help="Maximum peptide length")
     parser.add_argument('-ml', "--min_length", help="Minimum peptide length")
     parser.add_argument('-t', '--tools', help='Tools used for peptide predictions', required=True, type=str)
+    parser.add_argument('-sv', '--versions', help='File with used software versions.', required=True)
```
Comment (christopher-mohr) [inline bin/epaa.py:904]: Should we call it `methods` and maybe `File specifying the applied prediction methods`. We should also be consistent with single and double quotes.

File: main.nf [hunk 4]
```diff
@@ -286,7 +287,7 @@ process peptidePrediction {
    def ref_prot = params.proteome ? "--proteome ${params.proteome}" : ""
    def wt = params.wild_type ? "--wild_type" : ""
    """
-   epaa.py ${input_type} --identifier ${inputs.baseName} --alleles $alleles --mhcclass ${params.mhc_class} --max_length ${params.max_peptide_length} --min_length ${params.min_peptide_length} --tools ${params.tools} --reference ${params.genome} ${ref_prot} ${wt}
+   epaa.py ${input_type} --identifier ${inputs.baseName} --alleles $alleles --mhcclass ${params.mhc_class} --max_length ${params.max_peptide_length} --min_length ${params.min_peptide_length} --tools ${params.tools} --versions ${software_versions} --reference ${params.genome} ${ref_prot} ${wt}
```
Comment (christopher-mohr) [inline main.nf:290]: We have to adapt this in case we change the parameter name in `epaa.py`

File: bin/epaa.py (hunk 3 above)
Comment (skrakau) [reply bin/epaa.py:904]: I called it `--versions` to distinguish it from  `--tools`. The file contains the whole output from the process `get_software_versions`, independently if a tool specified with `--tool` or not.

Comment (skrakau) [reply bin/epaa.py]: Changed the quotes (also for `--tools` parameter).

File: main.nf (hunk 4 above)
Comment (skrakau) [reply main.nf:290]: see above

File: bin/epaa.py (hunk 1 above)
Comment (skrakau) [reply bin/epaa.py]: done

File: bin/epaa.py (hunk 2 above)
Comment (skrakau) [reply bin/epaa.py]: done

**Pull request merged**